bin/spice/cmd/chat.go

```diff
@@ -78,6 +79,13 @@ spice chat --model <model> --cloud
 			rtcontext.SetApiKey(apiKey)
 		}
 
+		userAgent, _ := cmd.Flags().GetString(userAgentKeyFlag)
```

Comment:
@slyons  - is my understanding correct that the target use case for this is to provide custom agent for E2E tests so we correctly handle such usage? Any other example where we would like tp override the agent?

Reply:
That's *our* use case for it, but it's a generic enough use case since the `User-Agent` will be exposed through the metrics